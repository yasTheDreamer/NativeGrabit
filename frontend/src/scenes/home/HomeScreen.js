import React from 'react';
import {View} from 'react-native';
import BackScene from '../../components/molecules/home_backscene/BackScene';
import Headline from '../../components/atoms/headline/Headline';
import SignButtons from '../../components/molecules/home_signButtons/signButtons';

const HomeScreen = ({navigation}) => {
  return (
    <View>
      <BackScene />
      <Headline />
      <SignButtons nav={navigation} />
    </View>
  );
};

export default HomeScreen;
