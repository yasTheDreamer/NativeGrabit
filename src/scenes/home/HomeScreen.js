import React from 'react';
import {View} from 'react-native';
import BackScene from '../../components/molecules/home_backscene/BackScene';
import Headline from '../../components/atoms/headline/Headline';
import SignButtons from '../../components/molecules/home_signButtons/signButtons';

const HomeScreen = () => {
  return (
    <View>
      <BackScene />
      <Headline />
      <SignButtons />
    </View>
  );
};

export default HomeScreen;
