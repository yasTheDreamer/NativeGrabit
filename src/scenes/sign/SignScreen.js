import React from 'react';
import {View, StyleSheet} from 'react-native';
import BackScene from '../../components/molecules/sign_backscene/BackScene';
import InfoOrganism from '../../components/organisms/infoOrganism/InfoOrganism';
import FacebookButton from '../../components/atoms/facebook/FacebookButton';

const SignScreen = () => {
  return (
    <View style={styles.container}>
      <BackScene />
      <InfoOrganism />
      <FacebookButton style={styles.facebookButton} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },

  facebookButton: {
    flexDirection: 'row',
    width: 300,
    height: 40,
  },
});

export default SignScreen;
