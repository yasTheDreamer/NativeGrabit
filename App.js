import React from 'react';
import {View} from 'react-native';
import HomeScreen from './src/scenes/home/HomeScreen';
import SignScreen from './src/scenes/sign/SignScreen';

const App = () => {
  return (
    <View>
      <SignScreen />
    </View>
  );
};

export default App;
