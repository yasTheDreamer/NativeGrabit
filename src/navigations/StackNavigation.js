import {createStackNavigator} from '@react-navigation/stack';
import HomeScreen from '../scenes/home/HomeScreen';
import SignScreen from '../scenes/sign/SignScreen';

const StackNavigator = createStackNavigator();

export default StackNavigator;
