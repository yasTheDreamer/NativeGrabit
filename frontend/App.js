import React from 'react';
import {StyleSheet} from 'react-native';
import HomeScreen from './src/scenes/home/HomeScreen';
import SignScreen from './src/scenes/sign/SignScreen';
import StackNavigation from './src/navigations/StackNavigation';
import {NavigationContainer} from '@react-navigation/native';

const App = () => {
  return (
    <NavigationContainer>
      <StackNavigation.Navigator headerMode={styles.navigator}>
        <StackNavigation.Screen name="Home" component={HomeScreen} />
        <StackNavigation.Screen name="Sign" component={SignScreen} />
      </StackNavigation.Navigator>
    </NavigationContainer>
  );
};

const styles = StyleSheet.create({
  navigator: {
    display: 'none',
  },
});

export default App;
