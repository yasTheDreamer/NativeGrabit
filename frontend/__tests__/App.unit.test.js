/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';
import {NavigationContainer} from '@react-navigation/native';
import StackNavigation from '../src/navigations/StackNavigation';
import HomeScreen from '../src/scenes/home/HomeScreen';
import SignScreen from '../src/scenes/sign/SignScreen';
import {render} from '@testing-library/react-native';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

// it('renders correctly', async () => {
//   const result = renderer.create(<App />);

//   await act(async () => {
//     expect(result).toMatchSnapshot();
//   });
// });

describe('App component', () => {
  const app = renderer.create(<App />).toJSON();
  const appRender = render(<App />);

  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('has only one child', () => {
    expect(app.children.length).toBe(1);
  });

  it('renders headline', () => {
    expect(appRender.findByText('Delivering Everything Else')).toBeTruthy();
  });

  describe('NavigatorContainer component', () => {
    const navigationContainer = renderer
      .create(
        <NavigationContainer testID="navigationContainer">
          <StackNavigation.Navigator
            testID="stackNavigator"
            headerMode={{display: 'none'}}>
            <StackNavigation.Screen name="Home" component={HomeScreen} />
            <StackNavigation.Screen name="Sign" component={SignScreen} />
          </StackNavigation.Navigator>
        </NavigationContainer>,
      )
      .toJSON();

    it('renders correctly', () => {
      expect(navigationContainer).toMatchSnapshot();
    });

    it('has only one child', () => {
      expect(navigationContainer.children.length).toBe(1);
    });


    

  });
});
