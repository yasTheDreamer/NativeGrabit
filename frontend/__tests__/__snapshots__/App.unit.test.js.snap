// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component NavigatorContainer component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "height": "100%",
                            "position": "relative",
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "testUri": "../../../src/assets/images/background.png",
                            }
                          }
                          style={
                            Object {
                              "height": "80%",
                              "resizeMode": "stretch",
                              "width": "100%",
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "height": 200,
                            "left": 0,
                            "position": "absolute",
                            "top": "30%",
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "testUri": "../../../src/assets/images/delivery.png",
                            }
                          }
                          style={
                            Object {
                              "height": "100%",
                              "width": "100%",
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 100,
                            "justifyContent": "center",
                            "left": 0,
                            "position": "absolute",
                            "top": 0,
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "testUri": "../../../src/assets/images/Logo.png",
                            }
                          }
                          style={Object {}}
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 100,
                            "left": 0,
                            "position": "absolute",
                            "top": "15%",
                            "width": "100%",
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#FFFFFF",
                              "fontFamily": "Roboto-Black",
                              "fontSize": 30,
                              "fontWeight": "500",
                              "textAlign": "center",
                              "width": "90%",
                            }
                          }
                        >
                          Delivering Everything Else
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "bottom": 0,
                            "flex": 1,
                            "height": 150,
                            "justifyContent": "center",
                            "position": "absolute",
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#333C45",
                              "borderRadius": 10,
                              "height": 45,
                              "justifyContent": "center",
                              "marginBottom": 10,
                              "opacity": 1,
                              "width": "90%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#FFFFFF",
                                "fontFamily": "Roboto-Bold",
                                "fontSize": 20,
                              }
                            }
                          >
                            Sign in
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#333C45",
                              "borderRadius": 10,
                              "height": 45,
                              "justifyContent": "center",
                              "marginBottom": 10,
                              "opacity": 1,
                              "width": "90%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#FFFFFF",
                                "fontFamily": "Roboto-Bold",
                                "fontSize": 20,
                              }
                            }
                          >
                            Sign up
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`App component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "height": "100%",
                            "position": "relative",
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "testUri": "../../../src/assets/images/background.png",
                            }
                          }
                          style={
                            Object {
                              "height": "80%",
                              "resizeMode": "stretch",
                              "width": "100%",
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "height": 200,
                            "left": 0,
                            "position": "absolute",
                            "top": "30%",
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "testUri": "../../../src/assets/images/delivery.png",
                            }
                          }
                          style={
                            Object {
                              "height": "100%",
                              "width": "100%",
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 100,
                            "justifyContent": "center",
                            "left": 0,
                            "position": "absolute",
                            "top": 0,
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "testUri": "../../../src/assets/images/Logo.png",
                            }
                          }
                          style={Object {}}
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 100,
                            "left": 0,
                            "position": "absolute",
                            "top": "15%",
                            "width": "100%",
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#FFFFFF",
                              "fontFamily": "Roboto-Black",
                              "fontSize": 30,
                              "fontWeight": "500",
                              "textAlign": "center",
                              "width": "90%",
                            }
                          }
                        >
                          Delivering Everything Else
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "bottom": 0,
                            "flex": 1,
                            "height": 150,
                            "justifyContent": "center",
                            "position": "absolute",
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#333C45",
                              "borderRadius": 10,
                              "height": 45,
                              "justifyContent": "center",
                              "marginBottom": 10,
                              "opacity": 1,
                              "width": "90%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#FFFFFF",
                                "fontFamily": "Roboto-Bold",
                                "fontSize": 20,
                              }
                            }
                          >
                            Sign in
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#333C45",
                              "borderRadius": 10,
                              "height": 45,
                              "justifyContent": "center",
                              "marginBottom": 10,
                              "opacity": 1,
                              "width": "90%",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#FFFFFF",
                                "fontFamily": "Roboto-Bold",
                                "fontSize": 20,
                              }
                            }
                          >
                            Sign up
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
